### PR TITLE
Skin menu selection

### PR DESCRIPTION
**Pull request by Codidodido**
- Create skin menu
- Select skin

### CODE REVIEW

File: Assets/Scripts/ChangeFrame.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+public class ChangeFrame : MonoBehaviour
+{
+    public Action onIncreaseFrame;
+    public Action onDecreaseFrame;
+
+    public void IncreaseFrame()
+    {
+        onIncreaseFrame?.Invoke();
+        Debug.Log("Increase");
```
Comment (URAmiRBin) [inline Assets/Scripts/ChangeFrame.cs]: remove the log messages

File: Assets/Scripts/ChangeFrame.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    public void DecreaseFrame()
+    {
+        onDecreaseFrame?.Invoke();
+        Debug.Log("Decrease");
```
Comment (URAmiRBin) [inline Assets/Scripts/ChangeFrame.cs]: also this

File: Assets/Scripts/ChangeFrame.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+using UnityEngine;
+
+public class ChangeFrame : MonoBehaviour
+{
+    public Action onIncreaseFrame;
+    public Action onDecreaseFrame;
+
+    public void IncreaseFrame()
+    {
+        onIncreaseFrame?.Invoke();
```
Comment (URAmiRBin) [inline Assets/Scripts/ChangeFrame.cs]: find a way to remove the null check

File: Assets/Scripts/SkinCardController.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    public void SetCard(Sprite skin, string name)
+    {
+        cardSkin.sprite = skin;
+        cardText.text = name;
+    }
+
+    public void SelectSkin()
+    {
+        PlayerPrefs.SetString("skin",cardSkin.sprite.name);
+        Debug.Log(cardSkin.sprite.name);
```
Comment (URAmiRBin) [inline Assets/Scripts/SkinCardController.cs]: no logs, just use debugger for test

File: Assets/Scripts/SkinCardDisplayer.cs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+{
+    [SerializeField] private Skins skins;
+    [SerializeField] private int skinCountPerRow;
+    [SerializeField] private GameObject cardsPrefab;
+    [SerializeField] private HorizontalLayoutGroup cardsGroup;
+    [Space] 
+    [SerializeField] private ChangeFrame increaseFrame;
+    [SerializeField] private ChangeFrame decreaseFrame;
+    
+    public int frame;
```
Comment (URAmiRBin) [inline Assets/Scripts/SkinCardDisplayer.cs]: frame is not an integer

File: Assets/Scripts/ChangeFrame.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using UnityEngine;
+
+public class ChangeFrame : MonoBehaviour
```
Comment (URAmiRBin) [inline Assets/Scripts/ChangeFrame.cs]: ChangeFrame is not a good name for this class

File: Assets/Scripts/SkinCardDisplayer.cs [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        increaseFrame.onIncreaseFrame -= IncreaseFrame;
+        decreaseFrame.onDecreaseFrame -= DecreaseFrame;
+    }
+
+    private void SpawnCards()
+    {
+        RemoveCards();
+        for (var i = frame; i < frame + skinCountPerRow; i++)
+        {
+            if(i<skins.SkinCards.Count)
```
Comment (URAmiRBin) [inline Assets/Scripts/SkinCardDisplayer.cs]:
```suggestion
            if (i < skins.SkinCards.Count)
```

for readablity

File: Assets/Scripts/SkinCardDisplayer.cs [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    private void CreateCard(SkinCard skinCard)
+    {
+        var card = Instantiate(cardsPrefab, Vector2.zero, Quaternion.identity);
+        card.transform.SetParent(cardsGroup.transform);
+        
+        card.GetComponent<SkinCardController>().SetCard(skinCard.skinSprite,skinCard.name);
+        
+        if(skinCard.skinSprite.name == PlayerPrefs.GetString("skin"))
```
Comment (URAmiRBin) [inline Assets/Scripts/SkinCardDisplayer.cs]: make this boolean a function, what does this do?

File: Assets/Scripts/SkinCardDisplayer.cs [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    private void RemoveCards()
+    {
+        var cards = cardsGroup.GetComponentsInChildren<SkinCardController>();
+        
+        foreach (var card in cards)
+        {
+            Destroy(card.gameObject);
+        }
```
Comment (URAmiRBin) [inline Assets/Scripts/SkinCardDisplayer.cs]:
```suggestion
            Destroy(card.gameObject);
```

**Review by URAmiRBin — CHANGES_REQUESTED** [review]

File: Assets/Scripts/SkinCardDisplayer.cs (hunk 8 above)
Comment (Codidodido) [reply Assets/Scripts/SkinCardDisplayer.cs]: Nothing That much important, I wanted to display selected card with red color, but I have to find better solution

File: Assets/Scripts/ChangeFrame.cs (hunk 3 above)
Comment (Codidodido) [reply Assets/Scripts/ChangeFrame.cs]: Why ?
Maybe I find solution easier and faster If I know the reason

File: Assets/Scripts/ChangeFrame.cs [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public Action onIncreaseFrame;
+    public Action onDecreaseFrame;
+
+    public void IncreaseFrame()
```
Comment (URAmiRBin) [inline Assets/Scripts/ChangeFrame.cs]: what does this mean??

**Review by URAmiRBin — CHANGES_REQUESTED** [review]

File: Assets/Scripts/ChangeFrame.cs (hunk 3 above)
Comment (URAmiRBin) [reply Assets/Scripts/ChangeFrame.cs]: null checks are bad code smell if you have learned your lessons well

File: Assets/Scripts/SkinCardDisplayer.cs (hunk 8 above)
Comment (URAmiRBin) [reply Assets/Scripts/SkinCardDisplayer.cs]: ok find a better solution

**Comment by URAmiRBin** [comment]
@Codidodido 
dont mark the conversations as resolved if you have not done any changes

File: Assets/Scripts/StartSkinCardIndexChanger.cs [hunk 11]
```diff
@@ -0,0 +1,21 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using UnityEngine;
+
+public class StartSkinCardIndexChanger : MonoBehaviour
```
Comment (URAmiRBin) [inline Assets/Scripts/StartSkinCardIndexChanger.cs]: What does this do the name does not ring any bells to me

File: Assets/Scripts/StartSkinCardIndexChanger.cs [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public Action onIncreaseFrame = () => { };
```
Comment (URAmiRBin) [inline Assets/Scripts/StartSkinCardIndexChanger.cs]: what does this mean? how do you increase frames?

**Review by URAmiRBin — CHANGES_REQUESTED** [review]

**Comment by URAmiRBin** [comment]
@Codidodido reviewed, still need some changes

**Review by URAmiRBin — APPROVED** [review]

**Pull request merged**